src/global_scheduler/global_scheduler.c

```diff
@@ -0,0 +1,116 @@
+#include <getopt.h>
+#include <signal.h>
+#include <stdlib.h>
+
+#include "common.h"
+#include "event_loop.h"
+#include "global_scheduler.h"
+#include "global_scheduler_algorithm.h"
+#include "net.h"
+#include "state/db_client_table.h"
+#include "state/table.h"
+#include "state/task_table.h"
+
+/* This is used to define the array of local schedulers used to define the
+ * global_scheduler_state type. */
+UT_icd local_scheduler_icd = {sizeof(local_scheduler), NULL, NULL, NULL};
+
+void assign_task_to_local_scheduler(global_scheduler_state *state,
+                                    task *original_task,
+                                    node_id node_id) {
+  task *updated_task =
+      alloc_task(task_task_spec(original_task), TASK_STATUS_SCHEDULED, node_id);
+  retry_info retry = {
+      .num_retries = 0, .timeout = 100, .fail_callback = NULL,
+  };
+  task_table_update(state->db, updated_task, &retry, NULL, NULL);
+  free_task(updated_task);
+}
+
+global_scheduler_state *init_global_scheduler(event_loop *loop,
+                                              const char *redis_addr,
+                                              int redis_port) {
+  global_scheduler_state *state = malloc(sizeof(global_scheduler_state));
+  state->db = db_connect(redis_addr, redis_port, "global_scheduler", "", -1);
```

Comment:
Yeah, I believe the handling is done inside of `db_connect`, e.g., https://github.com/ray-project/ray/blob/master/src/common/state/redis.c#L59